src/layouts/rails-component-layout.tsx

```diff
@@ -1,33 +1,253 @@
+      </>)
+    }
+  }
+
+  const renderSlots = (slots) => {
+    if (slots.length > 0) {
+      return(<>
+        <H2>Slots</H2>
+        {slots.map( (slot) => {
+          return(<>
+            <H3><InlineCode>{slot.name}</InlineCode></H3>
+            {/* @ts-ignore */}
+            <RailsMarkdown text={slot.description} parentRailsId={parentRailsId}/>
+            <PropsTable props={slot.parameters} parentRailsId={parentRailsId}/>
+          </>)
+        })}
+      </>)
+    }
+  }
+
+  const renderMethods = (methods) => {
+    if (methods.length > 0) {
+      return(<>
+        <H2>Methods</H2>
+        {methods.map( (method) => {
+          return(<>
+            <H3><InlineCode>{method.name}</InlineCode></H3>
+            {/* @ts-ignore */}
+            <RailsMarkdown text={method.description} parentRailsId={parentRailsId}/>
+            <PropsTable props={method.parameters} parentRailsId={parentRailsId}/>
+          </>)
+        })}
+      </>)
+    }
+  }
+
+  const renderPreviews = (previews) => {
+    if (showPreviews && previews.length > 0) {
+      return(<>
+        <H2>Examples</H2>
+        <LookbookEmbed height={300} previews={previews}/>
+      </>)
+    }
+  }
+
+  return(<>
+    {renderArguments(props)}
+    {renderSlots(slots)}
+    {renderMethods(methods)}
+    {renderPreviews(previews)}
+  </>)
+}
+
 export default function RailsComponentLayout({data}) {
+  const {name, short_name, a11y_reviewed, status, previews, slots} = data.railsComponent
+
   const title = data.sitePage?.context.frontmatter.title
   const description = data.sitePage?.context.frontmatter.description
-  const componentName = pascalCase(title)
+  const railsUrl = `${baseUrl}/components/${status}/${short_name.toLowerCase()}`
+  const reactId = data.sitePage.context.frontmatter.reactId
+  const railsId = data.sitePage.context.frontmatter.railsId
+  const figmaUrl = data.sitePage.context.frontmatter.figmaUrl
+
+  const subcomponents = []
+  const componentStack = [data.railsComponent]
+
+  while (componentStack.length > 0) {
+    const component = componentStack.pop()
+
+    if (component.subcomponents) {
+      subcomponents.push.apply(subcomponents, component.subcomponents)
+      componentStack.push.apply(componentStack, component.subcomponents)
+    }
+  }
+
+  const tableOfContents = {
+    items: [
+      {url: '#description', title: 'Description'},
+      {url: '#arguments', title: 'Arguments'},
+    ],
+  }
+
+  if (slots.length > 0) {
+    tableOfContents.items.push({url: '#slots', title: 'Slots'})
+  }
+
+  if (previews.length > 0) {
+    tableOfContents.items.push({url: '#examples', title: 'Examples'})
+  }
+
+  const slugger = new GithubSlugger()
+
+  for (const subcomponent of subcomponents) {
+    tableOfContents.items.push({
+      url: `#${slugger.slug((subcomponent as any).name)}`,
+      title: (subcomponent as any).name
+    })
+  }
+
+  const renderSubComponents = (subcomponents) => {
```

Comment:
In general, whenever you find yourself writing a function called `render*` in React, it's a good indication that it should be a component. I'd pull this out into a `RailsSubComponents` component